inc/lift/RequestStatus.h

```diff
@@ -19,7 +19,8 @@ enum class RequestStatus
     TIMEOUT,            ///< The request timed out.
     RESPONSE_EMPTY,     ///< The request has an empty response (socket severed).
 
-    ERROR               ///< The request had an error and failed.
+    ERROR,               ///< The request had an error and failed.
+    CURL_DOWNLOAD_ERROR  ///< The request had an error in the CURL write callback.
```

Comment:
this API uses curl internally but we don't expose any "curl" names through the API, so just make it DOWNLOAD_ERROR?

Reply:
Updated to use DOWNLOAD_ERROR